subsys/ipc/ipc_service/lib/icmsg.c

```diff
@@ -519,3 +528,22 @@ int icmsg_release_rx_buffer(const struct icmsg_config_t *conf,
 	return 0;
 }
 #endif /* CONFIG_IPC_SERVICE_ICMSG_NOCOPY_RX */
+
+#if IS_ENABLED(CONFIG_IPC_SERVICE_BACKEND_ICMSG_WQ_ENABLE)
+
+static int work_q_init(void)
```

Comment:
just a curiosity: why this is enabled by `SYS_INIT` and not as part of the init backend?

Reply:
There is a single workqueue shared among all backend instances. If you mean why not in `icmsg_open`, that would result in multiple initializations without some static flag. If you think a static flag is preferable, then I can change it to that.

Reply:
I see, not a problem but `SYS_INIT` is being currently reworked by @gmarull and I'm not sure what's in the future for that. @gmarull any opinion on this?

Reply:
SYS_INIT won't disappear anytime soon, so feel free to use it. In any case, if SYS_INIT is ever deprecated a replacement will exist.